ledger/block/src/helpers/target.rs

```diff
@@ -36,12 +36,12 @@ pub fn block_reward<N: Network>(
     time_since_last_block: i64,
     coinbase_reward: u64,
     transaction_fees: u64,
-) -> u64 {
+) -> Result<u64> {
     // Determine which block reward version to use.
-    match block_height < N::CONSENSUS_V2_HEIGHT {
-        true => block_reward_v1(total_supply, block_time, coinbase_reward, transaction_fees),
-        false => block_reward_v2(total_supply, time_since_last_block, coinbase_reward, transaction_fees),
-    }
+    Ok(match N::CONSENSUS_VERSION(block_height)? as usize {
```

Comment:
While the number of versions is small, I'd vote for matching on the enums for more clarity.

Reply:
I agree it would be clearer for now, but it will also create an *inevitable* moment of confusion to a future developer when they have to potentially refactor hundreds of lines of code to use a range comparison again. I vote for matching on integers still.